syntax = "proto3";
package trade;

message TradeRequest {
  string instrument = 1;
  string side = 2;
  int32 quantity = 3;
  double price = 4;
  string trader_id = 5;
}

message TradeResponse {
  string status = 1;
  string trade_id = 2;
  int32 latency_ms = 3;
}

service TradeService {
  rpc SubmitTrade (TradeRequest) returns (TradeResponse);
}
